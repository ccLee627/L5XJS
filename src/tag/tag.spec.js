const Tag = require("./index");

describe("Tag Class", () => {
    describe("New Instance", () => {
        it("Accepts Proper Input", () => {
            const fn = (
                name,
                datatype,
                desc = null,
                alias = null,
                safety = false,
                dim = null
            ) => () => new Tag(name, datatype, desc, alias, safety, dim);

            expect(fn("tagName")).toThrow();
            expect(fn(null, "DINT")).toThrow();
            expect(fn("tagName", "DINT")).not.toThrow();
            expect(fn("tagName", "DINT", 12)).toThrow();
            expect(fn("tagName", "DINT", "A Description")).not.toThrow();
            expect(fn("tagName", "DINT", "A Description", 12)).toThrow();
            expect(fn("tagName", "DINT", "A Description", "anotherTag")).not.toThrow();
            expect(fn("tagName", "DINT", "A Description", "anotherTag", 12)).toThrow();
            expect(fn("tagName", "DINT", "A Description", "anotherTag", true)).not.toThrow();
            expect(fn("tagName", "DINT", "A Description", "anotherTag", true, 12)).toThrow();
            expect(
                fn("tagName", "DINT", "A Description", "anotherTag", true, "[10]")
            ).not.toThrow();
        });

        it("Initializes with Desired Document Model", () => {
            let tag = new Tag("tagName", "DINT");
            expect(tag).toMatchSnapshot();

            tag = new Tag("tagName", "DINT", "A Description");
            expect(tag).toMatchSnapshot();

            tag = new Tag("tagName", "DINT", "A Description", "anotherTag");
            expect(tag).toMatchSnapshot();

            tag = new Tag("tagName", "DINT", "A Description", "anotherTag", true);
            expect(tag).toMatchSnapshot();

            tag = new Tag("tagName", "DINT", "A Description", "anotherTag", true, "[12]");
            expect(tag).toMatchSnapshot();
        });
    });

    describe("Static Methods", () => {
        test("isTag: Returns Appropriate Judgement", () => {
            let tag = new Tag("tagName", "DINT");
            expect(Tag.isTag(tag)).toBeTruthy();

            tag = { notATag: 12 };
            expect(Tag.isTag(tag)).toBeFalsy();

            tag = 12;
            expect(Tag.isTag(tag)).toBeFalsy();
        });
    });
});
