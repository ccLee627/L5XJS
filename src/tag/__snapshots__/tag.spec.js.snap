// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Class New Instance Initializes with Desired Document Model 1`] = `
Tag {
  "_dom": Object {
    "attributes": Object {
      "Class": "Standard",
      "DataType": "DINT",
      "Name": "tagName",
      "TagType": "Base",
    },
    "elements": Array [],
    "name": "Tag",
    "type": "element",
  },
  "class_id": "202108c6004705df7b5fd0add930619f",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Tag Class New Instance Initializes with Desired Document Model 2`] = `
Tag {
  "_dom": Object {
    "attributes": Object {
      "Class": "Standard",
      "DataType": "DINT",
      "Name": "tagName",
      "TagType": "Base",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "A Description",
            "type": "cdata",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Tag",
    "type": "element",
  },
  "class_id": "202108c6004705df7b5fd0add930619f",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Tag Class New Instance Initializes with Desired Document Model 3`] = `
Tag {
  "_dom": Object {
    "attributes": Object {
      "AliasFor": "anotherTag",
      "Class": "Standard",
      "DataType": "DINT",
      "Name": "tagName",
      "TagType": "Alias",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "A Description",
            "type": "cdata",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Tag",
    "type": "element",
  },
  "class_id": "202108c6004705df7b5fd0add930619f",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Tag Class New Instance Initializes with Desired Document Model 4`] = `
Tag {
  "_dom": Object {
    "attributes": Object {
      "AliasFor": "anotherTag",
      "Class": "Safety",
      "DataType": "DINT",
      "Name": "tagName",
      "TagType": "Alias",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "A Description",
            "type": "cdata",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Tag",
    "type": "element",
  },
  "class_id": "202108c6004705df7b5fd0add930619f",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;

exports[`Tag Class New Instance Initializes with Desired Document Model 5`] = `
Tag {
  "_dom": Object {
    "attributes": Object {
      "AliasFor": "anotherTag",
      "Class": "Safety",
      "DataType": "DINT",
      "Dimensions": "[12]",
      "Name": "tagName",
      "TagType": "Alias",
    },
    "elements": Array [
      Object {
        "elements": Array [
          Object {
            "cdata": "A Description",
            "type": "cdata",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Tag",
    "type": "element",
  },
  "class_id": "202108c6004705df7b5fd0add930619f",
  "elem_id": "33cb4159377b63ce9db11667160e39df",
}
`;
